tiledb/sm/query/result_tile.cc

```diff
@@ -266,6 +278,26 @@ bool ResultTile::same_coords(
   return true;
 }
 
+bool ResultTile::same_coords(uint64_t pos_a, uint64_t pos_b) const {
+  auto dim_num = coord_tiles_.size();
+  for (unsigned d = 0; d < dim_num; ++d) {
+    if (!domain_->dimension_ptr(d)->var_size()) {  // Fixed-sized
+      if (std::memcmp(coord(pos_a, d), coord(pos_b, d), coord_size(d)) != 0)
+        return false;
+    } else {  // Var-sized
+      if (coord_string(pos_a, d) != coord_string(pos_b, d))
+        return false;
+    }
+  }
+
+  return true;
+}
+
+uint64_t ResultTile::timestamp(uint64_t pos) {
+  const auto& tile = std::get<0>(*this->tile_tuple(constants::timestamps));
+  return tile.data_as<uint64_t>()[pos];
```

Comment:
It is optional, but as this is going to be called on a very tight loop, I'm hesitant to add extra checks. Unfortunately, the way ResultTile is built at the moment requires this to be optional :(

Reply:
Can you please explain for my understanding why it is required to be optional?

In any case, even if it wasn't optional, we should at least be sure `pos` it's within `tile` boundaries. If you still think this will have a significant performance cost, can we open a ticket, and perhaps add a comment here referencing it, about checking the impact while we are doing our perf testing of the feature.

I see that all those issues come from the way `ResultTile` class is designed/used and the fact it's 2-step initialized etc. Maybe eventually we should also target a refactor of that class, it seems a quite had task but if you have seen defects in this area in the past, maybe it's worth considering it.

Reply:
We don't check pos is in the tile boundary for the zipped_coord or unzipped_coord functions which are doing the same kind of operations and called from the same code that calls timestamp. I think in this case it's fine to put the responsibility on the caller to pass valid arguments as this is cell by cell processing and very performance sensitive.

The reason it is optional is because ResultTiles are more often than not used for fragments that will not be consolidated with timestamps, adding in a static Tile to the class will bloat the size of ResultTile and increase memory usage unnecessarily. We will end up making this class c.41 compliant, but that will require all class contained within to be c.41 compliant and all containers using ResultTiles to be changed to something that would accept having the contained objects to possibly be not created, so that will take time :)

Reply:
Thanks for the detailed explanation !